src/init.jl

```diff
@@ -95,17 +99,22 @@ function __init__()
         #     return v, g
         # end
 
-        function Hamiltonian(metric, ℓπ)
+        function Hamiltonian(metric::AbstractMetric, ℓπ, ::Union{ForwardDiffAD, Type{ForwardDiffAD}})

```

Comment:
I would decide on either a type or an instance (see https://docs.julialang.org/en/v1/manual/style-guide/#Avoid-confusion-about-whether-something-is-an-instance-or-a-type-1). According to the style guide preferably an instance, I guess.

Reply:
I will fix this.